### PR TITLE
Add AssetManager

### PR DESCRIPTION
**Pull request by windy1**
[**API**](https://github.com/SpongePowered/SpongeAPI/pull/1126) | [Common](https://github.com/SpongePowered/SpongeCommon/pull/558) | [Forge](https://github.com/SpongePowered/SpongeForge/pull/562) | [Vanilla](https://github.com/SpongePowered/SpongeVanilla/pull/230) | [Gradle](https://github.com/SpongePowered/SpongeGradle/pull/3) | [plugin-meta](https://github.com/Minecrell/plugin-meta/pull/1)

This PR adds the `AssetManager` class to Sponge. It allows for easy retrieval of plugin resources as well as adds a `assets` plugin meta attribute to allow developers to declare a custom asset directory.

Signed-off-by: Walker Crouse <walkercrouse@hotmail.com>

### CODE REVIEW

File: src/main/java/org/spongepowered/api/asset/AssetManager.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    /**
+     * Returns the {@link Asset} of the specified name for the specified
+     * {@link PluginContainer}.
+     *
+     * @param plugin Plugin container
+     * @param name Name of resource to retrieve
+     * @return Asset if present, empty otherwise
+     */
+    Optional<Asset> getAsset(PluginContainer plugin, String name);
```
Comment (simon816) [inline src/main/java/org/spongepowered/api/asset/AssetManager.java]: This isn't needed as the method below can just accept the PluginContainer as an object

Comment (windy1) [inline src/main/java/org/spongepowered/api/asset/AssetManager.java]: Ah okay, thanks.

**Comment by JBYoshi** [comment]
Wow, a six-repo pull request! Never seen that before :smile:

**Comment by kinggoesgaming** [comment]
@JBYoshi you jinxed it :sad:

**Comment by windy1** [comment]
@kinggoesgaming It was merged :P

**Pull request closed without merging**